python.d/python_modules/bases/FrameworkServices/SocketService.py

```diff
@@ -249,6 +273,28 @@ def _parse_config(self):
             except (KeyError, TypeError):
                 self.debug('No port specified. Using: "{0}"'.format(self.port))
 
+            self.tls = bool(self.configuration.get('tls', self.tls))
+            if self.tls and not _TLS_SUPPORT:
+                self.warning('TLS requested but no TLS module found, disabling TLS support.')
+                self.tls = False
+            if _TLS_SUPPORT and not self.tls:
+                self.debug('No TLS preference specified, not using TLS.')
+
+            if self.tls and _TLS_SUPPORT:
+                self.key = self.configuration.get('tls_key_file')
+                self.cert = self.configuration.get('tls_cert_file')
+                if not self.cert:
+                    # If there's not a valid certificate, clear the key too.
+                    self.debug('No valid TLS client certificate configuration found.')
+                    self.key = None
+                    self.cert = None
+                elif not self.key:
+                    # If a key isn't listed, the config may still be
+                    # valid, because there may be a key attached to the
+                    # certificate.
+                    self.notice('No TLS client key specified, assuming it\'s attached to the certificate.')
```

Comment:
`self.notice` ?

Reply:
Strictly speaking it's not an error as far as the `ssl` module is concerned, as long as there's a key attached to the certificate file (which is as simple as appending the PEM encoded data to the end of the certificate file).  Unfortunately, there's not really any easy way for us to validate that there is a key attached to the certificate (unless you want me to put a bare-bones PEM parser in here).

Reply:
i mean what is `self.notice` ? logger has no `notice` method